core/templates/dev/head/app.js

```diff
@@ -355,6 +355,62 @@ oppia.factory('windowDimensionsService', ['$window', function($window) {
   };
 }]);
 
+// Service for sending events to Google Analytics.
+//
+// Note that events are only sent if the CAN_SEND_ANALYTICS_EVENTS flag is
+// turned on. This flag must be turned on explicitly by the application
+// owner.
+oppia.factory('siteAnalyticsService', ['$window', function($window) {
+  var CAN_SEND_ANALYTICS_EVENTS = GLOBALS.CAN_SEND_ANALYTICS_EVENTS;
+
+  // For definitions of the various arguments, please see:
+  // developers.google.com/analytics/devguides/collection/analyticsjs/events
+  var _sendEventToGa = function(eventCategory, eventAction, eventLabel) {
+    if ($window.ga && CAN_SEND_ANALYTICS_EVENTS) {
+      $window.ga('send', 'event', eventCategory, eventAction, eventLabel);
+    }
+  };
+
+  // For definitions of the various arguments, please see:
+  // developers.google.com/analytics/devguides/collection/analyticsjs/
+  //   social-interactions
+  var _sendSocialEventToGa = function(network, action, targetUrl) {
+    if ($window.ga && CAN_SEND_ANALYTICS_EVENTS) {
+      $window.ga('send', 'social', network, action, targetUrl);
+    }
+  };
+
+  return {
+    // The srcElement refers to the element on the page that is clicked.
+    registerStartLoginEvent: function(srcElement) {
+      _sendEventToGa(
+        'LoginButton', 'click', $window.location.pathname + ' ' + srcElement);
+    },
+    registerOpenExplorationCreationModalEvent: function() {
```

Comment:
No, I think that what I implemented is pretty robust. What I was objecting to is just measuring the click on the button on the page directly, because although the button looks visually the same in both cases, it actually does different things altogether. I felt that it would be better to treat these two cases as distinct events.

In other words, clicks on 'create exploration' trigger different events depending on whether the user is logged in or not, because those clicks end up doing different things: in one case, it starts the login process, and in another case, it actually brings up the modal. Hence the current structure. Please let me know what you think about it -- is it fine, or are changes needed?